src/operator/single.ts

```diff
@@ -55,20 +52,25 @@ class SingleSubscriber<T> extends Subscriber<T> {
 
   protected _next(value: T): void {
     const predicate = this.predicate;
-    const currentIndex = this.index++;
-
+    this.index++;
     if (predicate) {
-      let result = tryCatch(predicate)(value, currentIndex, this.source);
-      if (result === errorObject) {
-        this.destination.error(errorObject.e);
-      } else if (result) {
-        this.applySingleValue(value);
-      }
+      this.__tryNext(value);
     } else {
       this.applySingleValue(value);
     }
   }
 
+  private __tryNext(value: T): void {
```

Comment:
I think single underbar would work for consistency of this changes, other `tryxxx` does that. I agree using underbar in codebase sometimes not clear - I generally try to avoid to use it. In cases like conflict such as `public next(): void` and `protected _next(): void`, each encapsulated function would start with underbar, but never for public interfaces.